### PR TITLE
add info on scaling logic to typha overview

### PR DESCRIPTION
**Pull request by jgilfoil**
This information wasn't readily available anywhere in the docs that i could find, so I wanted to make it easier to find for users.

Product Version(s): 3.27

Issue: No issue on this that i'm aware of

Link to docs preview: ~Sorry, I don't know how to find this.~
https://deploy-preview-1360--tigera.netlify.app/calico/latest/reference/typha/overview

SME review:
- [ ] An SME has approved this change. 

DOCS review:
- [ ] A member of the docs team has approved this change.

Additional information:

Merge checklist:

- [ ] Deploy preview inspected wherever changes were made 
- [ ] Build completed successfully
- [ ] Test have passed

### CODE REVIEW

**Comment by jgilfoil** [comment]
@tigera/docs

**Comment by lwr20** [comment]
Looks good.  

I had a conversation with the other maintainers about whether we should document this implementation detail or not.  Its a balance.  The conclusion was that this table is fine, but it would be good to add a caveat that "we may change the algorithm as needs evolve".  For example, at high scale we might reduce the number of typhas in future.

**Comment by lwr20** [comment]
Also, once we're happy with the text, we should copy it into `calico/reference/typha/overview.mdx` (so that the v3.28 versioned docs will get this change).

**Comment by lwr20** [comment]
I've added a [link](https://deploy-preview-1360--tigera.netlify.app/calico/latest/reference/typha/overview) to the preview page so you can see your change as it will appear on the website.

**Comment by jgilfoil** [comment]
Thanks for the feedback. I've added a note about potential for change and added the update to the `calico/reference/typha/overview.mdx` file as well.

File: calico/reference/typha/overview.mdx [hunk 1]
```diff
@@ -16,3 +16,31 @@ If you are using the Kubernetes API Datastore, we recommend using Typha. Althoug
 
 - Since one Typha instance can support hundreds of Felix instances, it reduces the load on the datastore by a large factor.
 - Since Typha can filter out updates that are not relevant to Felix, it also reduces Felix's CPU usage. In a high-scale (100+ node) Kubernetes cluster, this is essential because the number of updates generated by the API server scales with the number of nodes.
+
+## Operator Scaling Logic
```
Comment (ctauchen) [inline calico/reference/typha/overview.mdx:20]:
```suggestion
## Operator scaling logic
```
We use sentence case for headings.

File: calico/reference/typha/overview.mdx [hunk 2]
```diff
@@ -16,3 +16,31 @@ If you are using the Kubernetes API Datastore, we recommend using Typha. Althoug
...
+
+If you're using the Operator installation, then the operator will automatically scales Typha pods for optimal performance.
```
Comment (ctauchen) [inline calico/reference/typha/overview.mdx:22]:
```suggestion
If you installed {{prodname}} using the operator, then the operator will automatically scale Typha pods to optimize performance. 
```

File: calico/reference/typha/overview.mdx [hunk 3]
```diff
@@ -16,3 +16,31 @@ If you are using the Kubernetes API Datastore, we recommend using Typha. Althoug
 
 - Since one Typha instance can support hundreds of Felix instances, it reduces the load on the datastore by a large factor.
 - Since Typha can filter out updates that are not relevant to Felix, it also reduces Felix's CPU usage. In a high-scale (100+ node) Kubernetes cluster, this is essential because the number of updates generated by the API server scales with the number of nodes.
+
+## Operator Scaling Logic
+
+If you're using the Operator installation, then the operator will automatically scales Typha pods for optimal performance. 
+
+- Small Cluster Considerations: Special handling is in place for clusters with fewer than five nodes.
```
Comment (ctauchen) [inline calico/reference/typha/overview.mdx:24]: * It's not clear what this special handling is.
* Whatever it is, we can keep the explanation as a regular sentence in the same paragraph as the previous sentence. For example, "For clusters with fewer than five nodes, we do things like this."
* If the special handling is merely what's described in the table, let's just let the table do the work. We can strike out this line.

File: calico/reference/typha/overview.mdx [hunk 4]
```diff
@@ -16,3 +16,31 @@ If you are using the Kubernetes API Datastore, we recommend using Typha. Althoug
...
+
+Here's the scaling logic it uses:
```
Comment (ctauchen) [inline calico/reference/typha/overview.mdx:26]: Remove. The table explains itself.

File: calico/reference/typha/overview.mdx [hunk 5]
```diff
@@ -16,3 +16,31 @@ If you are using the Kubernetes API Datastore, we recommend using Typha. Althoug
...
+| `...` |   |
+| `<2000` | 12 |
+| `...` |   |
+| `<3600` | 20 |
+
+:::note
+
+This algorithm may change as needs evolve.
+
+:::
```
Comment (ctauchen) [inline calico/reference/typha/overview.mdx:46]: On second thought, let's remove this too. The same applies to pretty much everything in the docs. If we make a change, that should be documented just like any other change.

**Review by ctauchen — COMMENTED** [review]
@jgilfoil A few final changes for you. Some nits, a few comments and suggested deletions. Let me know if you have any questions. All comments apply to both files.

**Pull request closed without merging**